riff-raff/app/views/validation/validationErrors.scala.html

```diff
@@ -3,8 +3,19 @@
 
 @main("Validation errors", request) {
     <div class="clearfix"><p>&nbsp;</p></div>
-    @errors.errors.map { error =>
-        Error: @error
+    @for((context, errors) <- errors.errors.toList.groupBy(_.context)) {
+        <div class="panel panel-danger">
+            <div class="panel-heading">
+                <h3 class="panel-title"><span class="glyphicon glyphicon-exclamation-sign" aria-hidden="true"></span><span>Error in @context</span></h3>
+            </div>
+            <div class="panel-body">
+                <ul>
+                @errors.map(_.message).map { msg =>
+                  <li>@msg</li>
+                }
```

Comment:
Indeed. This is clearly the kind of thing that happens when refactoring code at midnight.